### DISCUSSION TITLE
When building for production, I get "Unable to locate a Java Runtime."

### DISCUSSION THREAD

**Discussion by athelene** [quasarframework/quasar#13541] [CLI - Capacitor / Cordova modes]
I'm not sure if I missed something on installation. I have been prompted to update Gradle a few times but didn't based on the information at this[ link. ](https://quasar.dev/quasar-cli-webpack/developing-capacitor-apps/build-commands). I've searched the Quasar docs and can't find any reference to Java runtime. Didn't need it for the dev build. Can't find information on what or where to install this.

**Comment by higuitadiaz** [comment]
Hi, could you send a snapshot?

Sometimes I had problems with Cordova and I found that problem is with the Android version or Android Studio.
As I don't have a reference of the error in your project, I don't know how to help you. 

My suggestion is:

Verify that your Android studio is correct.
1. Create a blank quasar project
2. Add Cordova mode
3. Run the script on Cordova.

If everything is OK, then we can focus on your project, if not, reinstall Android Studio. (I'm assuming you're on Windows)

Finally, Capacitor work so fine too and is compatible with Cordova plugins

**Comment by athelene** [comment]
After more research, I tried using the following commands to see if there was a path error: 

 echo $ANDROID_SDK_ROOT

and 

 echo $ANDROID_HOME

Neither of these returned anything. So it looked to me like JAVA didn't exist at all. (Wouldn't this have been installed when I installed quasar and/or Android Studio?) So I started looking around to see if I could find it anywhere. Interestingly, I see that my library/android/sdk folder is empty. Seems like maybe it should be there? If that's the case, what do I do? Not sure of what version I should use with Quasar and the Gradle it provided. I searched the Quasar/Capacitor docs again and don't see anything about installing JAVA. I'm using a Mac if it makes any difference.

**Reply by MilosPaunovic** [reply]
Start by installing Java, like described [here](https://www.java.com/en/download/help/mac_install.html).

**Comment by athelene** [comment]
After following MilosPaunovic's link and installing Java, I got past that error.  Yeah! However, now I'm getting the following. Looks like I'm still missing something? I'm really confused as I didn't see any instructions on the Quasar site that indicated I had to do other installs before using their product. If I missed it, it would be helpful if someone could point me in the right direction. Or do they just assume you have/know what other products are required?  

> Configure project :app
google-services.json not found, google-services plugin not applied. Push Notifications won't work

> Task :capacitor-cordova-android-plugins:compileReleaseJavaWithJavac FAILED

FAILURE: Build failed with an exception.

**Reply by metalsadman** [reply]
it's in the logs, you are using a plugin that asks for a configuration json file.

**Comment by athelene** [comment]
Wouldn't Capacitor be expected to create whatever configuration file is required? I thought the point of Capacitor was to take a web app and convert it and create files as required for Android or iOS? If I'm wrong about that, I would love to know where to find the documentation on setting up the google-service.json file?

**Reply by higuitadiaz** [reply]
No, capacitor only simulates a Web within an app and acts as a gateway between the device and the Web App.

The google-service.json file goes in the path src-capacitor/android/app however this will important according to the service to use (Sample: push-notification / App Check / etc...) but It doesn't make sense to use it for firebase, because it goes in your Web App and not in Capacitor

https://capacitorjs.com/docs/guides/push-notifications-firebase
